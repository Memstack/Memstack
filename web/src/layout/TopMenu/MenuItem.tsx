import React from "react";
import { Link } from "react-router-dom";

interface MenuItemProps {
  icon: string;
  to?: string;
  text?: string;
}

const MenuItem: React.SFC<MenuItemProps> = ({ icon, to, text }) => (
  <Link to={to || "#"}>
    <div className={"item"}>
      <span className="icon">
        <i className={icon} />
      </span>
      <span className={"menu-text"}>{text}</span>
    </div>
  </Link>
);

export default MenuItem;
