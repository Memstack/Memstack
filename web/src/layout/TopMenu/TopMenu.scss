@import "../../scss/index.scss";

.Navbar {
  position: sticky;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 30;

  background-color: $navbar-color;

  .secondary {
    width: 100%;
    background-color: $primary-color;
  }

  .icon {
    margin-left: 0;
    padding-right: 0;
    @include media-query(small) {
      margin-left: 7px;
      padding-right: 0.5rem;
    }
  }

  .account-icon {
    margin-right: 0;
    @include media-query(small) {
      margin-right: 5px;
    }
  }

  .menu-text {
    float: right;
  }

  a,
  a:hover {
    color: white !important;
    text-decoration: none;
  }

  .main {
    width: 100%;

    height: $navbar-mobile-height;
    @include media-query(small) {
      height: $navbar-height;
    }
    display: flex;

    justify-content: center;

    @include media-query(small) {
      justify-content: space-between;
    }

    img {
      height: 1.75rem;
      min-width: 1.75rem;
      align-self: center;
    }

    .right {
      display: none;

      @include media-query(small) {
        display: flex;
      }
    }

    .item {
      height: 100%;
      padding: 0 0.85rem;
      align-items: center;
      display: flex;
      color: white;

      &:hover {
        background-color: #2b2c2e;
        transition: background 1s ease;
      }
    }
  }
}
