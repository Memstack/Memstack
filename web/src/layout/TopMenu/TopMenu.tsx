import React from "react";
import Logo from "../../logo.png";
import "./TopMenu.scss";
import { Link } from "react-router-dom";
import MenuItem from "./MenuItem";

interface ITopMenuState {
  isActive: boolean;
}

const actionMenuItems = [
  { to: "/add-card", text: "", icon: "fas fa-plus", extraClass: "" },
  { to: "/", text: "", icon: "fas fa-book-open", extraClass: "" },
  { to: "/explore", text: "", icon: "fas fa-chart-bar", extraClass: "" }
];
const menuItems = [
  { to: "/account", text: "Account", icon: "fas fa-user", extraClass: "" },
  { to: "/settings", text: "Settings", icon: "fas fa-cog", extraClass: "" }
];

class TopMenu extends React.Component<{}, ITopMenuState> {
  state = {
    isActive: false
  };

  toggleNav = () => {
    this.setState(prevState => ({
      isActive: !prevState.isActive
    }));
  };

  render() {
    const location = new URL(window.location.href);

    return (
      <nav className="Navbar">
        <div className="main">
          <div className="logo">
            <Link to="/">
              <div className="item">
                <img src={Logo} alt="Memstack" />
              </div>
            </Link>
          </div>
          <div className="menu left">
            {actionMenuItems.map((item, idx) => (
              <MenuItem
                icon={item.icon}
                to={item.to}
                text={item.text}
                key={idx}
              />
            ))}
          </div>

          <div className="menu right">
            {menuItems.map((item, idx) => (
              <MenuItem
                icon={item.icon}
                to={item.to}
                text={item.text}
                key={idx}
              />
            ))}
            <a
              href={`https://memstack-dev.auth.eu-west-2.amazoncognito.com/login?response_type=code&client_id=3pe23g9hu3rs1a4c9ua7ic8ebh&redirect_uri=https://${
                location.host
              }/login`}
            >
              <div className={"item"}>
                <span className="icon">
                  <i className={"fas fa-user"} />
                </span>
                <span className={"menu-text"}>Login</span>
              </div>
            </a>
          </div>
        </div>
      </nav>
    );
  }
}

export default TopMenu;
