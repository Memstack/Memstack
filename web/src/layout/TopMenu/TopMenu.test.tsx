import { mount, shallow } from "enzyme";
import React from "react";
import TopMenu from "./TopMenu";
import { MemoryRouter } from "react-router";

describe("TopMenu", () => {
  it("renders without crashing", () => {
    shallow(<TopMenu />);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<TopMenu />);
    expect(wrapper).toMatchSnapshot();
  });

  it("sets active to true when burger is clicked", () => {
    const component = mount(
      <MemoryRouter>
        <TopMenu />
      </MemoryRouter>
    );
    const burger = component.find(".burger");

    const stateBeforeClick = (component.find(TopMenu).instance() as TopMenu)
      .state.isActive;
    burger.simulate("click");
    const stateAfterClick = (component.find(TopMenu).instance() as TopMenu)
      .state.isActive;

    expect(stateBeforeClick).toBe(false);
    expect(stateAfterClick).toBe(true);
  });
});
