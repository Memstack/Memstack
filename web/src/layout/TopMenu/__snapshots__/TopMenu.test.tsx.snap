// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenu matches snapshot 1`] = `
<nav
  className="Navbar"
>
  <div
    className="main"
  >
    <div
      className="left"
    >
      <Link
        replace={false}
        to="/"
      >
        <div
          className="item"
        >
          <img
            src="logo.png"
          />
        </div>
      </Link>
    </div>
    <div
      className="right"
    >
      <Link
        key="0"
        replace={false}
        to="/account"
      >
        <div
          className="item"
        >
          <span
            className="icon"
          >
            <i
              className="fas fa-user"
            />
          </span>
          <span
            className="menu-text"
          >
            Account
          </span>
        </div>
      </Link>
      <Link
        key="1"
        replace={false}
        to="/settings"
      >
        <div
          className="item"
        >
          <span
            className="icon"
          >
            <i
              className="fas fa-cog"
            />
          </span>
          <span
            className="menu-text"
          >
            Settings
          </span>
        </div>
      </Link>
    </div>
    <div
      className="burger mobile-only"
      onClick={[Function]}
    >
      <span
        className=""
      />
      <span
        className=""
      />
      <span
        className=""
      />
    </div>
  </div>
  <div
    className="dropdown"
  >
    <Link
      key="Account"
      replace={false}
      to="/account"
    >
      <div
        className="item"
      >
        <span
          className="icon"
        >
          <i
            className="fas fa-user"
          />
        </span>
        <span
          className="menu-text"
        >
          Account
        </span>
      </div>
    </Link>
    <Link
      key="Settings"
      replace={false}
      to="/settings"
    >
      <div
        className="item"
      >
        <span
          className="icon"
        >
          <i
            className="fas fa-cog"
          />
        </span>
        <span
          className="menu-text"
        >
          Settings
        </span>
      </div>
    </Link>
    <Link
      key="Decks"
      replace={false}
      to="/"
    >
      <div
        className="item secondary"
      >
        <span
          className="icon"
        >
          <i
            className="fas fa-server"
          />
        </span>
        <span
          className="menu-text"
        >
          Decks
        </span>
      </div>
    </Link>
    <Link
      key="Explore"
      replace={false}
      to="/"
    >
      <div
        className="item secondary"
      >
        <span
          className="icon"
        >
          <i
            className="fas fa-chart-bar"
          />
        </span>
        <span
          className="menu-text"
        >
          Explore
        </span>
      </div>
    </Link>
    <Link
      key="Settings"
      replace={false}
      to="/"
    >
      <div
        className="item secondary"
      >
        <span
          className="icon"
        >
          <i
            className="fas fa-cog"
          />
        </span>
        <span
          className="menu-text"
        >
          Settings
        </span>
      </div>
    </Link>
  </div>
</nav>
`;
